### PR TITLE
[#58865] add enterprise banner to new custom field form

### PR DESCRIPTION
**Pull request by Kharonus**
# Ticket
[OP#58865](https://community.openproject.org/work_packages/58865)

# What are you trying to accomplish?
- do not allow creation of hierarchy custom fields if no enterprise token is active

# What approach did you choose and why?
- show banner below form field selection, if
  - hierarchy is selected AND
  - enterprise token is not active
- disable submit button, if the enterprise banner is shown

### CODE REVIEW

File: app/views/custom_fields/_form.html.erb [hunk 1]
```diff
@@ -62,6 +59,17 @@ See COPYRIGHT and LICENSE files for more details.
                  }
     %>
   </div>
+  <div class="form--field-extra-actions" <%= format_dependent.attributes(:enterpriseBanner) %>>
+    <%= angular_component_tag "opce-enterprise-banner",
+                              inputs: {
+                                collapsible: true,
+                                opReferrer: "custom-field-hierarchy",
+                                textMessage: "PLACEHOLDER - upgrade to Starkiller Base",
```
Comment (dominic-braeunlein) [inline app/views/custom_fields/_form.html.erb:67]: Hi @Kharonus, here is still a placeholder UI copy in the PR.

**Review by dominic-braeunlein — CHANGES_REQUESTED** [review]

File: app/helpers/custom_fields_helper.rb [hunk 2]
```diff
@@ -198,21 +198,25 @@ def custom_field_formats_for_select(custom_field)
...
 
   def label_for_custom_field_format(format_string)
-    format = OpenProject::CustomFieldFormat.find_by_name(format_string)
+    show_enterprise_text = format_string == "hierarchy" && !EnterpriseToken.allows_to?(:custom_field_hierarchies)
 
-    if format
-      format.label.is_a?(Proc) ? format.label.call : I18n.t(format.label)
-    end
+    format = OpenProject::CustomFieldFormat.find_by(name: format_string)
+    label = if format
```
Comment (apfohl) [inline app/helpers/custom_fields_helper.rb]: 🟡 Could be rewritten to `format&.label.is_a?(Proc)`.

**Review by apfohl — APPROVED** [review]
One small thing you can irgnore.

File: app/views/custom_fields/_form.html.erb (hunk 1 above)
Comment (Kharonus) [reply app/views/custom_fields/_form.html.erb:67]: Yes, there is a [TASK](https://community.openproject.org/wp/59219) for Maya, to provide the correct text.

My take was, as this is behind a feature flag, we can merge it even with the placeholder. But before removing the flag, we need to replace it.

File: app/helpers/custom_fields_helper.rb (hunk 2 above)
Comment (Kharonus) [reply app/helpers/custom_fields_helper.rb]: rewritten with a bit of reordering...

**Pull request merged**